src/SFML/Window/Win32/WindowImplWin32.cpp

```diff
@@ -1003,14 +1021,78 @@ void WindowImplWin32::processEvent(UINT message, WPARAM wParam, LPARAM lParam)
             if ((wParam == DBT_DEVICEARRIVAL) || (wParam == DBT_DEVICEREMOVECOMPLETE))
             {
                 // Some sort of device change has happened, update joystick connections if it is a device interface
-                auto* deviceBroadcastHeader = reinterpret_cast<DEV_BROADCAST_HDR*>(lParam);
+                const auto* deviceBroadcastHeader = reinterpret_cast<DEV_BROADCAST_HDR*>(lParam);
 
                 if (deviceBroadcastHeader && (deviceBroadcastHeader->dbch_devicetype == DBT_DEVTYP_DEVICEINTERFACE))
                     JoystickImpl::updateConnections();
             }
 
             break;
         }
+        case WM_TOUCH:
+        {
+            // Get the number of events
+            const unsigned int count = LOWORD(wParam);
+
+            // Reserve memory
+            std::vector<TOUCHINPUT> events(count);
+
+            if (GetTouchInputInfo(reinterpret_cast<HTOUCHINPUT>(lParam), count, events.data(), sizeof(TOUCHINPUT)))
+            {
+                POINT point;
+                for (unsigned int i = 0; i < count; ++i)
```

Comment:
Could this be a ranged for loop?

Reply:
Agree, range-based for was also in my snippet. Two more suggestions I made were not applied:
* Early exit condition first -- instead of having 50 LoC with extra indentation, check if `GetTouchInputInfo` returns 0 and already leave the scope.
* Explicitly check against 0, since we are dealing with integers, not `bool`s.

The whole logic inside `case WM_TOUCH` might even deserve its own function, which could clarify things with scope and `break` a bit.